### PR TITLE
test: migrate params tests to vitest

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Consolidate the testing frameworks and migrate to vitest.

**Description**

Migrate `params` tests to vitest.

Closes #issue_number

- Run all tests

### CODE REVIEW

File: packages/params/test/unit/applicationDomains.test.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ describe("validate application domains", () => {
       for (let i = 0; i < DOMAIN_APPLICATION_MASK.length; i++) {
         r += DOMAIN_APPLICATION_MASK[i] & domain[i];
       }
-      expect(r).to.be.above(0, `${name} mask application should be valid`);
+      expect(r).toBeGreaterThan(0);
```
Comment (nflaig) [inline packages/params/test/unit/applicationDomains.test.ts]: could keep comment, from the assertion it is not instantly obvious that this means the mask is valid

File: packages/params/package.json [hunk 2]
```diff
@@ -52,10 +52,13 @@
...
-    "test": "yarn run check-types",
-    "test:unit": "mocha 'test/unit/**/*.test.ts'",
-    "test:browsers": "yarn karma start karma.config.cjs",
-    "test:e2e": "LODESTAR_PRESET=minimal mocha 'test/e2e/**/*.test.ts'",
+    "test": "yarn run check-types",    
+    "test:unit": "vitest --run --dir test/unit/ --coverage",
+    "test:browsers": "yarn test:browsers:chrome && yarn test:browsers:firefox && yarn test:browsers:electron",
+    "test:browsers:chrome": "vitest --run --browser chrome --config ./vitest.browser.config.ts --dir test/unit",
+    "test:browsers:firefox": "vitest --run --browser firefox --config ./vitest.browser.config.ts --dir test/unit",
+    "test:browsers:electron": "echo 'Electron tests will be introduced back in the future as soon vitest supports electron.'",
```
Comment (nflaig) [inline packages/params/package.json:60]: just a question regarding this message, did we have electron tests in the past? If not, might wanna rephrase this

**Review by nflaig — APPROVED** [review]

File: packages/params/package.json (hunk 2 above)
Comment (nflaig) [reply packages/params/package.json:60]: @nazarhussain ☝️

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.14.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.14.0) :tada:

**Pull request merged**